crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -122,11 +127,43 @@ impl AuthorityStorePruner {
         total_pruned
     }
 
+    fn handle_checkpoint(
+        checkpoint_execution_state: CheckpointExecutionState,
+        objects: &DBMap<ObjectKey, Object>,
+    ) -> anyhow::Result<usize> {
+        let mut pruned = 0;
+        let mut wb = objects.batch();
+        let mut updates = HashMap::new();
+
+        for effects in checkpoint_execution_state.effects {
+            for (object_id, seq_number) in effects.modified_at_versions {
```

Comment:
highlighting this piece for verification
FWIU `modified_at_versions` should do the job for live pruner case

Reply:
@phoenix-o I suggested down below to write a test to ensure this. I think it would be worth having such a test.

Reply:
sure, just double checking that `modified_at_versions` is the right aggregator because there might be some edge cases that won't be handled by such test either (i.e. wrapped objects, dynamic fileds, etc.) But so far it looks like it's still the right container

Reply:
Yep that's correct. For `unwrapped_then_deleted` operations, the modified at version for that object will be the version it was wrapped at, which was already deleted. So you would be doing a double delete. That, I think, is the only relevant edge case if you're just processing `modified_at_versions`.

Would it work for your use case to just do a best effort deletion? If the obj version doesn't exist, ignore and move on? If not, I am working on https://github.com/MystenLabs/sui/pull/8012 to separate these operations in effects, which could tell you which ones to skip

Reply:
double delete is fine, pruning is idempotent
is there any edge case you have in mind? 
otherwise it sounds like `modified_at_versions` container is good enough for pruning

Reply:
Yeah then in that case I think you're good